### PR TITLE
Add specs for `system` macro method

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: spec/compiler/macro/macro_methods_spec.cr [hunk 1]
```diff
@@ -10,6 +10,37 @@ private def declare_class_var(container : ClassVarContainer, name, var_type : Ty
...
+private def exit_code_command(code)
+  {% if flag?(:win32) %}
+    %(cmd.exe /c "exit #{code}")
+  {% else %}
+    case code
+    when 0
+      "true"
+    when 1
+      "false"
+    else
```
Comment (beta-ziliani) [inline spec/compiler/macro/macro_methods_spec.cr:22]: Why this distinction?

Comment (straight-shoota) [reply spec/compiler/macro/macro_methods_spec.cr:22]: Not sure. I just copied this helper 1:1 from `process_spec`.

Comment (straight-shoota) [reply spec/compiler/macro/macro_methods_spec.cr:22]: I don't think this is really necessary, but I also didn't want to touch that existing code. We might want to refactor it anyway (cf #12775).

**Review by beta-ziliani — APPROVED** [review]

**Review by sdogruyol — APPROVED** [review]
Thank you @straight-shoota 🙏

**Pull request merged**